frontend/lib/ui/test_results_page/test_results_filters_view.dart

```diff
@@ -47,78 +47,24 @@ class _TestResultsFiltersViewState
   static const double _comboWidth = 260;
   static const double _controlHeight = 48;
 
-  late Set<String> _families;
-  late Set<String> _envs;
-  late Set<String> _tests;
-  late Set<(String, String)> _executionMetadata;
-  late DateTime? _fromDate;
-  late DateTime? _untilDate;
+  late TestResultsFilters _selectedFilters;
 
   @override
   void initState() {
     super.initState();
-    _loadFromUri();
+    _selectedFilters = widget.initialFilters;
   }
 
   @override
   void didUpdateWidget(covariant TestResultsFiltersView oldWidget) {
     super.didUpdateWidget(oldWidget);
-    if (oldWidget.pageUri.toString() != widget.pageUri.toString()) {
-      _loadFromUri();
-      setState(() {});
+    if (oldWidget.initialFilters != widget.initialFilters) {
+      setState(() {
+        _selectedFilters = widget.initialFilters;
+      });
     }
   }
 
-  void _loadFromUri() {
-    Set<String> readParam(String key) {
-      final params = widget.pageUri.queryParametersAll;
-      final raw = params[key] ?? const <String>[];
-      return raw.expand((s) => s.split(',')).where((s) => s.isNotEmpty).toSet();
-    }
-
-    DateTime? readDateParam(String key) {
-      final params = widget.pageUri.queryParametersAll;
-      final raw = params[key] ?? const <String>[];
-      return raw
-          .map((s) => DateTime.tryParse(s))
-          .firstOrNullWhere((d) => d != null);
-    }
-
-    _families = readParam('families');
```

Comment:
Since query parameter parsing is now confined to the `TestResultsFilters` model, we don't need duplicate parsing code both here and `testResultsSearchProvider`. This is easier to update in the future when adding more parameters.